fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/OxidizableBlocksRegistry.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.registry;
+
+import java.util.Optional;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Oxidizable;
+import net.minecraft.item.HoneycombItem;
+
+import net.fabricmc.fabric.api.util.OxidizableFamily;
+
+public class OxidizableBlocksRegistry {
+	private static final BiMap<Block, Block> OXIDIZATION_LEVEL_INCREASES = HashBiMap.create();
+	private static final BiMap<Block, Block> OXIDIZATION_LEVEL_DECREASES = HashBiMap.create();
+	private static final BiMap<Block, Block> UNWAXED_TO_WAXED_BLOCKS = HashBiMap.create();
+	private static final BiMap<Block, Block> WAXED_TO_UNWAXED_BLOCKS = HashBiMap.create();
+
+	static {
+		// Put all vanilla entries in here, just in case!
+		OXIDIZATION_LEVEL_INCREASES.putAll(Oxidizable.OXIDATION_LEVEL_INCREASES.get());
+		OXIDIZATION_LEVEL_DECREASES.putAll(Oxidizable.OXIDATION_LEVEL_DECREASES.get());
+		UNWAXED_TO_WAXED_BLOCKS.putAll(HoneycombItem.UNWAXED_TO_WAXED_BLOCKS.get());
+		WAXED_TO_UNWAXED_BLOCKS.putAll(HoneycombItem.WAXED_TO_UNWAXED_BLOCKS.get());
+	}
+
+	/**
+	 * Registers multiple {@link OxidizableFamily}s.
+	 * @param families the families to register
+	 */
+	public static void registerFamilies(OxidizableFamily... families) {
+		for (OxidizableFamily family : families) {
+			registerFamily(family);
+		}
+	}
+
+	/**
+	 * Registers an {@link OxidizableFamily}s.
+	 * @param family the {@link OxidizableFamily} to register
+	 */
+	public static void registerFamily(OxidizableFamily family) {
+		OXIDIZATION_LEVEL_INCREASES.putAll(family.oxidizationLevelIncreasesMap());
+		OXIDIZATION_LEVEL_DECREASES.putAll(family.oxidizationLevelDecreasesMap());
+		UNWAXED_TO_WAXED_BLOCKS.putAll(family.unwaxedToWaxedMap());
+		WAXED_TO_UNWAXED_BLOCKS.putAll(family.waxedToUnwaxedMap());
+	}
+
+	/**
+	 * Registers a waxable block.
+	 * Unnecessary if part of a registered {@link OxidizableFamily}.
+	 * @see #registerFamily(OxidizableFamily)
+	 * @param unwaxed the unwaxed variant
+	 * @param waxed the waxed variant
+	 */
+	public static void registerWaxable(Block unwaxed, Block waxed) {
```

Comment:
This function and many others could use some `Objects.requireNonNull` assertions.

Reply:
Good point, I never really use that myself but I should...